syntax = "proto3"; 

message InnerInner {
    repeated string values = 1;
}

message InnerList {
    repeated InnerInner lists = 1;
}

message OuterList {
    repeated InnerList cards = 1;
}


message DictCards {
    map<int32, OuterList> gamedict = 1;
}


message Tuple {
    string first = 1;
    string second = 2;
}

message MovesList {
    repeated Tuple move = 1;
}

message Game { 
    repeated string players = 1;
    repeated MovesList moves = 2;
    repeated int32 result = 3;
    bool running = 4; 
    int32 rand_num = 5; 
    int32 start_counter = 6;
    int32 random_num_counter = 7;
}